UefiPayloadPkg/UefiPayloadPkg.dsc

```diff
@@ -188,7 +189,11 @@
   TimerLib|UefiPayloadPkg/Library/AcpiTimerLib/AcpiTimerLib.inf
   ResetSystemLib|UefiPayloadPkg/Library/ResetSystemLib/ResetSystemLib.inf
   SerialPortLib|MdeModulePkg/Library/BaseSerialPortLib16550/BaseSerialPortLib16550.inf
+!if $(UNIVERSAL_PAYLOAD) == TRUE
+  PlatformHookLib|UefiPayloadPkg/Library/UniversalPayloadPlatformHookLib/UniversalPayloadPlatformHookLib.inf
+!else
   PlatformHookLib|UefiPayloadPkg/Library/PlatformHookLib/PlatformHookLib.inf
+!endif

```

Comment:
For upstream, we had better check if we could re-use same library instance for both cases.
we could upstream this change once the final decision is made.